pgmpy/estimators/ConstraintBasedEstimator.py

```diff
@@ -0,0 +1,648 @@
+
+        >>> # pdag_to_dag is static:
+        ... pdag1 = DirectedGraph([('A', 'B'), ('C', 'B'), ('C', 'D'), ('D', 'C'), ('D', 'A'), ('A', 'D')])
+        >>> ConstraintBasedEstimator.pdag_to_dag(pdag1).edges()
+        [('D', 'C'), ('C', 'B'), ('A', 'B'), ('A', 'D')]
+
+        >>> # example of a pdag with no faithful extension:
+        ... pdag2 = DirectedGraph([('A', 'B'), ('A', 'C'), ('B', 'C'), ('C', 'B')])
+        >>> ConstraintBasedEstimator.pdag_to_dag(pdag2).edges()
+        UserWarning: PDAG has no faithful extension (= no oriented DAG with the same v-structures as PDAG).
+        Remaining undirected PDAG edges oriented arbitrarily.
+        [('B', 'C'), ('A', 'B'), ('A', 'C')]
+        """
+
+        dag = BayesianModel()
+        dag.add_nodes_from(pdag.nodes())
+
+        # add already directed edges of pdag to dag
+        for X, Y in pdag.edges():
+            if not pdag.has_edge(Y, X):
+                dag.add_edge(X, Y)
+
+        while pdag.number_of_nodes() > 0:
+            # find node with (1) no directed outgoing edges and
+            #                (2) the set of undirected neighbors is either empty or
+            #                    undirected neighbors + parents of X are a clique
+            found = False
+            for X in pdag.nodes():
+                directed_outgoing_edges = set(pdag.successors(X)) - set(pdag.predecessors(X))
+                undirected_neighbors = set(pdag.successors(X)) & set(pdag.predecessors(X))
+                neighbors_are_clique = all((pdag.has_edge(Y, Z)
+                                            for Z in pdag.predecessors(X)
+                                            for Y in undirected_neighbors if not Y == Z))
+
+                if not directed_outgoing_edges and \
+                        (not undirected_neighbors or neighbors_are_clique):
+                    found = True
+                    # add all edges of X as outgoing edges to dag
+                    for Y in pdag.predecessors(X):
+                        dag.add_edge(Y, X)
+
+                    pdag.remove_node(X)
+                    break
+
+            if not found:
+                warn("PDAG has no faithful extension (= no oriented DAG with the " +
+                     "same v-structures as PDAG). Remaining undirected PDAG edges " +
+                     "oriented arbitrarily.")
+                for X, Y in pdag.edges():
+                    if not dag.has_edge(Y, X):
+                        try:
+                            dag.add_edge(X, Y)
+                        except ValueError:
+                            pass
+                break
+
+        return dag
+
+    @staticmethod
+    def model_to_pdag(model):
+        """Construct the DAG pattern (representing the I-equivalence class) for
+        a given BayesianModel. This is the "inverse" to pdag_to_dag.
+        """
+
+        if not isinstance(model, BayesianModel):
+            TypeError("'model' must be a BayesianModel()-instance.")
```

Comment:
@chrisittner There should be ```raise TypeError```.
And I think the message ```"model: Expected BayesianModel instance, got type {model_type}".format(model_type=type(model))"``` would be consistent with other TypeErrors.